x-pack/plugins/cases/public/containers/use_get_case_file_stats.tsx

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { UseQueryResult } from '@tanstack/react-query';
+
+import { useFilesContext } from '@kbn/shared-ux-file-context';
+import { useQuery } from '@tanstack/react-query';
+
+import type { Owner } from '../../common/constants/types';
+import type { ServerError } from '../types';
+
+import { constructFileKindIdByOwner } from '../../common/constants';
+import { useCasesToast } from '../common/use_cases_toast';
+import { useCasesContext } from '../components/cases_context/use_cases_context';
+import { casesQueriesKeys } from './constants';
+import * as i18n from './translations';
+
+export interface GetCaseFileStatsParams {
+  caseId: string;
+}
+
+export const useGetCaseFileStats = ({
+  caseId,
+}: GetCaseFileStatsParams): UseQueryResult<{ total: number }> => {
+  const { owner } = useCasesContext();
+  const { showErrorToast } = useCasesToast();
+  const { client: filesClient } = useFilesContext();
+
+  return useQuery(
+    casesQueriesKeys.caseFileStats(caseId),
+    () => {
+      return filesClient.list({
```

Comment:
Can we use the `select` property of react query to format the response to return only `{ total: number }`?

Reply:
Good idea, will do 👌